src/main/java/uk/gov/di/ipv/lambda/UserIdentityHandler.java

```diff
@@ -0,0 +1,78 @@
+package uk.gov.di.ipv.lambda;
+
+import com.amazonaws.services.lambda.runtime.Context;
+import com.amazonaws.services.lambda.runtime.RequestHandler;
+import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyRequestEvent;
+import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyResponseEvent;
+import com.nimbusds.oauth2.sdk.ErrorObject;
+import com.nimbusds.oauth2.sdk.OAuth2Error;
+import com.nimbusds.oauth2.sdk.ParseException;
+import com.nimbusds.oauth2.sdk.token.AccessToken;
+import com.nimbusds.oauth2.sdk.util.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.gov.di.ipv.helpers.ApiGatewayResponseGenerator;
+import uk.gov.di.ipv.helpers.RequestHelper;
+import uk.gov.di.ipv.service.AccessTokenService;
+import uk.gov.di.ipv.service.UserIdentityService;
+import uk.gov.di.ipv.validation.ValidationResult;
+
+import java.util.Map;
+
+public class UserIdentityHandler implements RequestHandler<APIGatewayProxyRequestEvent, APIGatewayProxyResponseEvent> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(UserIdentityHandler.class);
+    private static final String AUTHORIZATION_HEADER_KEY = "Authorization";
+
+    private final UserIdentityService userIdentityService;
+    private final AccessTokenService accessTokenService;
+
+    public UserIdentityHandler(UserIdentityService userIdentityService, AccessTokenService accessTokenService) {
+        this.userIdentityService = userIdentityService;
+        this.accessTokenService = accessTokenService;
+    }
+
+    public UserIdentityHandler() {
+        this.userIdentityService = new UserIdentityService();
+        this.accessTokenService = new AccessTokenService();
+    }
+
+    @Override
+    public APIGatewayProxyResponseEvent handleRequest(APIGatewayProxyRequestEvent input, Context context) {
+        ValidationResult<ErrorObject> validationResult = validateRequest(input.getHeaders());
+
+        if (!validationResult.isValid()) {
+            LOGGER.error(validationResult.getError().getDescription());
+            return ApiGatewayResponseGenerator.proxyJsonResponse(validationResult.getError().getHTTPStatusCode(), validationResult.getError().toJSONObject());
+        }
+
+        try {
+            String accessTokenString = RequestHelper.getHeaderByKey(input.getHeaders(), AUTHORIZATION_HEADER_KEY);
+
+            AccessToken.parse(accessTokenString);
```

Comment:
Leave comment to explain why we are using .parse() - what kind of validation is it doing etc?